src/simplewallet/simplewallet.cpp

```diff
@@ -3778,6 +3792,22 @@ bool simple_wallet::set_daemon(const std::vector<std::string>& args)
     }
     LOCK_IDLE_SCOPE();
     m_wallet->init(daemon_url);
+    if (trustedness)
+      m_trusted_daemon = trustedness;
```

Comment:
Aside from the default-to-trusted-or-untrusted question, I'm a bit concerned with this idiom of `boost::optional<uint32_t>` which seems to make the code look more complicated than necessary. Can't it be simply like this?

```c++
  if (args.size() == 2)
  {
    if (args[1] == "trusted")
      m_trusted_daemon = true;
    else if (args[1] == "untrusted")
      m_trusted_daemon = false;
    else
    {
      fail_msg_writer() << tr("Expected trusted or untrusted, got ") << args[1];
      return true;
    }
  }
  else
  {
    m_trusted_daemon = false;
    try
    {
      if (tools::is_local_address(m_wallet->get_daemon_address()))
      {
        MINFO(tr("Daemon is local, assuming trusted"));
        m_trusted_daemon = true;
      }
    }
    catch (const std::exception &e) { }
    success_msg_writer() << boost::format("Daemon set to %s, %s") % daemon_url % (*m_trusted_daemon ? tr("trusted") : r("untrusted"));
  }
```

If the current idiom is to be used, I think this line should be `m_trusted_daemon = *trustedness`.

Reply:
It should be optional<bool> fixed. The proposed change is either before wallet2::init, in which case get_daemon_address is invalid, and so I'd have to add parsing code, or after it, in which case if the keyword is wrong and we have to error out, it's already set. Guess I can avoid erroring out, and set it to untrusted instead.

Reply:
As for the * or no *, it is equivalent in practice here AFAICT.

Reply:
Done, I does save a couple lines

Reply:
er, I squashed to the wrong patch, gimme a min...